agent/agent_endpoint.go

```diff
@@ -141,7 +141,7 @@ func (s *HTTPServer) AgentReload(resp http.ResponseWriter, req *http.Request) (i
 	}
 
 	// Trigger the reload
-	errCh := make(chan error, 0)
+	errCh := make(chan error)
```

Comment:
Either we should keep the `0` capacity here or add a comment about this chan being purposefully unbufferred so someone in the future wont come along and not know what was intended.

Reply:
Correct me if I'm wrong here, but 0 is the default capacity of a channel, so it's not changing anything - except omitting the default value. I think that just makes it a bit more readable, because whenever I see "make(chan foo, X)" at first glance I assume it's a buffered channel (just to then be surprised by the 0 when looking closer).

Reply:
Yeah, 0 / unbufferred is the default for sure but for me or others coming back to read the same bit of code seeing a zero there is a small indicator that the original author intended the channel to not be buffered instead of just not having thought about it and used a default.

I guess in this case its already pretty clear from the code that it doesn't need buffering as it will immediately wait on the chan and in the context of how its used its pretty easy to understand that only 1 thing is ever expected to come through the chan.

Reply:
@mkeeler Thanks for that intriguing insight! It's actually a rather good point you make, and it didn't occur to me that this notion may convey that extra piece of information to the reader. Will certainly keep that in mind.